src/core/filehandle.cpp

```diff
@@ -54,7 +54,8 @@ FileHandle::FileHandle(const char *filename, const char *mode, int error_source,
 }
 
 FileHandle::~FileHandle() {
-	avio_close(avio);
+	if (avio != nullptr)
```

Comment:
no need for this

`
int avio_close(AVIOContext *s)
{
    AVIOInternal *internal;
    URLContext *h;

    if (!s)
        return 0;
`